### ISSUE TITLE
For Cycle 9

### ISSUE THREAD

**Issue by stejbac** [bisq-network/compensation#459]
## Summary

> _Specify the total amount of BSQ you are requesting:_

 - BSQ requested: 1250 / 0.79 + 1700 / 0.81 + 2000 / 0.67 = 6666.12 BSQ
 
 

## Contributions delivered

> _Provide links to contributions you have [delivered](https://github.com/bisq-network/proposals/issues/19), the amount of BSQ you are requesting for each, and any comments that will help stakeholders understand its value._

#### I missed the Cycle 7 and 8 deadlines for the following PRs:
- [Fix invalid comparators](https://github.com/bisq-network/bisq/pull/3251) (450 USD)
- [Fix proposal table hide when switching to another screen and back](https://github.com/bisq-network/bisq/pull/3356) (300 USD)
- [Fix failing test when there are spaces in the project path](https://github.com/bisq-network/bisq/pull/3578) (150 USD)
- [Prevent tooltip popover flicker upon mouseover](https://github.com/bisq-network/bisq/pull/3211) (350 USD)

#### I missed the Cycle 8 deadline for the following PRs:
- [Avoid unchecked casts/assignments/calls](https://github.com/bisq-network/bisq/pull/3626) (900 USD)
- [Un-generify Popup class](https://github.com/bisq-network/bisq/pull/3651) (150 USD)
- [Prevent taking of offers with unequal bank account types (excl. SEPA)](https://github.com/bisq-network/bisq/pull/3673) (300 USD)
- [Fix #3662: Malformed trade chat & dispute speech bubbles](https://github.com/bisq-network/bisq/pull/3680) (300 USD)

#### Merged PRs for Cycle 9:
- [Add transient tx map to DaoState to speed up getTx queries](https://github.com/bisq-network/bisq/pull/3773) (1200 USD)
- [Remove unused method parameters](https://github.com/bisq-network/bisq/pull/3806) (0 USD)
- [Fix doc comment warnings in TradeWalletService & OfferUtil](https://github.com/bisq-network/bisq/pull/3812) (0 USD)
- [Fix temp file path memory leak](https://github.com/bisq-network/bisq/pull/3817) (400 USD)
- [Speed up trades charts view load](https://github.com/bisq-network/bisq/pull/3828) (400 USD)

#### Other work (meant for Cycle 8 but missed):
- Spotted issue and commented on https://github.com/bisq-network/bisq/pull/3658 and https://github.com/bisq-network/bisq/pull/3661 (50 USD)
- Reported speech bubble issue https://github.com/bisq-network/bisq/issues/3662 fixed above (0 USD)

## Contributions in progress

> _Provide links to work you're involved with that is still [in progress](https://github.com/bisq-network/proposals/issues/19). **This section is optional,** and is for your own benefit in keeping track of what you're doing and keeping other contributors up to date with the same._

- [Use keyset delta algorithm to speed up GetData requests](https://github.com/bisq-network/bisq/pull/3896)

## Roles performed

> _Provide links to your [monthly report](https://github.com/bisq-network/proposals/issues/13) on any roles you are responsible for._

**Comment by cbeams** [comment]
@stejbac, this has been my first chance to review some of your work, and I just want to say it was a pleasure looking through your pull requests, commit comments and code changes. Keep going!

**Comment by cbeams** [comment]
Having a second look at this, I realize I overlooked this line item:

> - [Avoid unchecked casts/assignments/calls](bisq-network/bisq#3626) (900 USD)

That seems a bit much to me. Can you explain why you valued it so highly?

I will still vote to approve this request, but under the budget we'll roll out in Cycle 10, pure refactoring like this probably wouldn't get green lit at such a price.

**Comment by stejbac** [comment]
It was mainly because it affected a lot of files and took me quite a while (about 5 days), since I tried to cover the bulk of the codebase and some of the unchecked operations were not all that straightforward to replace.

I intended to include the item in Cycle 8 but missed the submission in the app (as I left it until very late, forgetting that blocks come randomly). So I just carried forward all the BSQ/dollar requests meant for Cycles 7 & 8 (at which time I was unaware of the new policy).

In future, I'll try to avoid making pure refactoring PRs and try to do them along the way instead.
